specification/communication/resource-manager/Microsoft.Communication/preview/2023-03-01-preview/SenderUsernames.json

```diff
@@ -0,0 +1,333 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "EmailServices.json#/parameters/EmailServiceName"
+          },
+          {
+            "$ref": "Domains.json#/parameters/DomainName"
+          },
+          {
+            "$ref": "#/parameters/SenderUsername"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describe the corresponding SenderUsername resource.",
+            "schema": {
+              "$ref": "#/definitions/SenderUsernameResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "Get SenderUsernames resource": {
+            "$ref": "./examples/senderUsernames/get.json"
+          }
+        }
+      },
+      "put": {
+        "description": "Add a new SenderUsername resource under the parent Domains resource or update an existing SenderUsername resource.",
+        "operationId": "SenderUsernames_CreateOrUpdate",
+        "summary": "Create Or Update",
+        "tags": [
+          "SenderUsernames"
+        ],
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Parameters for the create or update operation",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SenderUsernameResource"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "EmailServices.json#/parameters/EmailServiceName"
+          },
+          {
+            "$ref": "Domains.json#/parameters/DomainName"
+          },
+          {
+            "$ref": "#/parameters/SenderUsername"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "The create or update request was completed successfully",
```

Comment:
Only updates should come with 200 status. Creates should come back with 201 status.

Reply:
You return 201 to indicate the resource is newly created. NOT to indicate that it is a long-running operation. Whether it is a long-running operation or no is indicated by the provisioningState property having a non-terminal status when you return 200 or 201.

Reply:
> You return 201 to indicate the resource is newly created. NOT to indicate that it is a long-running operation. Whether it is a long-running operation or no is indicated by the provisioningState property having a non-terminal status when you return 200 or 201.

This makes sense. I have added the 201 response.